spec/acts_as_tenant/model_extensions_spec.rb

```diff
@@ -461,4 +463,98 @@
       expect(ActsAsTenant.current_tenant).to eq(@account)
     end
   end
+
+  describe 'overlapping tenants' do
+    before(:each) do
+      ActsAsTenant.without_tenant do
+        @account1 = Account.create!
+        @account2 = Account.create!
+        @project1 = Project.create!(name: 'Project1')
+        @project2 = Project.create!(name: 'Project2')
+      end
+
+      ActsAsTenant.with_tenant(@account1) do
+        @manager1 = Manager.create!(name: 'Manager1')
+      end
+
+      ActsAsTenant.with_tenant(@project1) do
+        @manager2 = Manager.create!(account: @account1, name: 'Manager2')
+        @manager1.update!(project_id: @project1.id)
+      end
+
+      ActsAsTenant.with_tenant(@project2) do
+        @manager3 = Manager.create!(account: @account1, name: 'Manager3')
+      end
+    end
+
+    context 'when current tenant is the first tenant defined for this model' do
+      it 'correctly scopes to the first tenant' do
+        expect(ActsAsTenant.with_tenant(@account1) { Manager.count }).to eq(3)
+        expect(ActsAsTenant.with_tenant(@account2) { Manager.count }).to eq(0)
+      end
+    end
+
+    context 'when current tenant is the second tenant defined for this model' do
+      it 'correctly scopes to the second tenant' do
+        expect(ActsAsTenant.with_tenant(@project1) { Manager.count }).to eq(2)
+        expect(ActsAsTenant.with_tenant(@project2) { Manager.count }).to eq(1)
+
+        expect(ActsAsTenant.with_tenant(@project1) { Manager.first }).to eq(@manager1)
+        expect(ActsAsTenant.with_tenant(@project1) { Manager.second }).to eq(@manager2)
+        expect(ActsAsTenant.with_tenant(@project2) { Manager.first }).to eq(@manager3)
+      end
+    end
+
+    context 'when creating a new model' do
+      before do
+        ActsAsTenant.current_tenant = @project1
+        @manager4 = Manager.create!(account: @account1, name: 'Manager4')
+        @manager5 = Manager.create!(name: 'Manager5')
+      end
+
+      it 'scopes it to overlapping tenants' do
+        expect(ActsAsTenant.with_tenant(@project1) { Manager.find_by(id: @manager4.id)}).not_to be_nil
+        expect(ActsAsTenant.with_tenant(@project2) { Manager.find_by(id: @manager4.id)}).to be_nil
+        expect(ActsAsTenant.with_tenant(@account1) { Manager.find_by(id: @manager4.id)}).not_to be_nil
+
+        expect(ActsAsTenant.with_tenant(@project1) { Manager.find_by(id: @manager5.id)}).not_to be_nil
+        expect(ActsAsTenant.with_tenant(@account1) { Manager.find_by(id: @manager5.id)}).to be_nil
+      end
+    end
+
+    context 'when trying to overwrite tenant on existing model' do
+      it 'should raise an error' do
+        expect { @manager1.update!(project_id: @project2.id) }.to raise_error ActsAsTenant::Errors::TenantIsImmutable
+        expect { @manager1.update!(account_id: @account2.id) }.to raise_error ActsAsTenant::Errors::TenantIsImmutable
+        expect { @manager1.update!(project_id: @account1.id) }.not_to raise_error

```

Comment:
why are you setting the `project_id` to an `@account.id`?

Reply:
also isn't `@manager1`s project tenant already set?

Reply:
This is to test that assigning the same tenant id is allowed (otherwise you won't be able to save the nested tenant model)

Reply:
lol the only reason this passed was cause `@account1` and `@project1` had the same ids